### PR TITLE
Reborn - Upsell modal pixel polish

### PR DESCRIPTION
**Pull request by thomasrogerlux**
### ❓ Context

- **Impacted projects**: `live-mobile`, `live-common`
  
- **Linked resource(s)**: [LIVE-2277], [figma designs](https://www.figma.com/proto/kbcqFhPS0O1GQ4zPoXpGJa/LLM-%2F-Wallet?page-id=5838%3A110374&node-id=5838%3A110538&viewport=985%2C560%2C0.37&scaling=scale-down&starting-point-node-id=5838%3A110538)

Pixel polish the upsell modal (the one asking you to purchase a nano during onboarding or during use of the app without a nano setup). Update live common to add new props to the IconBox component for shape and variant.

### ✅ Checklist

- [ ] **Test coverage**: _Did you write any tests to cover the changes introduced by this pull request?_
- [x] **Atomic delivery**: _Is this pull request standalone? In order words, does it depend on nothing else?_
- [x] **No breaking changes**: _Does this pull request contain breaking changes of any kind? If so, please explain why._

### 📸 Demo

Screenshot_20220525-175609_LL  DEV

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-2277]: https://ledgerhq.atlassian.net/browse/LIVE-2277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ui/packages/native/src/components/Icon/IconBox/index.tsx [hunk 1]
```diff
@@ -5,6 +5,8 @@ import Flex, { FlexBoxProps } from "../../Layout/Flex";
 
 type Props = {
   Icon: IconOrElementType;
+  variant?: "outlined" | "plain";
```
Comment (nparigi-ledger) [inline libs/ui/packages/native/src/components/Icon/IconBox/index.tsx:8]: The IconBox component is deprecated :grimacing: 
You have to modify BoxedIcon instead ... which already has a round variant
Sorry, there is a lack of documentation on this ... we should have deleted IconBox already

File: apps/ledger-live-mobile/src/screens/BuyDeviceScreen.tsx [hunk 2]
```diff
@@ -91,83 +93,110 @@ export default function BuyDeviceScreen() {
 
   const videoMounted = !useIsAppInBackground();
 
+  const isDuringOnboarding = () => {
+    const routes = navigation.getState().routes;
+    const prevRoute = routes[routes.length - 2];
+
+    return prevRoute?.name === ScreenName.OnboardingDoYouHaveALedgerDevice;
```
Comment (nparigi-ledger) [inline apps/ledger-live-mobile/src/screens/BuyDeviceScreen.tsx:100]: Isn't there a longer term way to do this that won't break as soon as we add a new page in the flow?

File: libs/ui/packages/native/src/components/Icon/IconBox/index.tsx (hunk 1 above)
Comment (thomasrogerlux) [reply libs/ui/packages/native/src/components/Icon/IconBox/index.tsx]: Done

File: apps/ledger-live-mobile/src/screens/BuyDeviceScreen.tsx (hunk 2 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/BuyDeviceScreen.tsx]: Done with redux

**Review by nparigi-ledger — APPROVED** [review]

**Pull request merged**